qutip/tests/core/data/test_operators_hypothesis.py

```diff
@@ -0,0 +1,83 @@
+import numpy
+
+from hypothesis import given, strategies as st
+
+from qutip.core import data as _data
+from qutip.tests import strategies as qst
+
+
+@given(qst.qobj_np(), qst.qobj_dtypes())
+def test_data_create(np_array, dtype):
+    if len(np_array.shape) == 1:
+        np_array = numpy.atleast_2d(np_array).transpose()
+    else:
+        np_array = numpy.atleast_2d(np_array)
+    data = _data.to(dtype, _data.create(np_array))
+    numpy.testing.assert_allclose(
+        data.to_array(), numpy.atleast_2d(np_array), rtol=1e-15
+    )
+
+
+@given(qst.qobj_datas())
+def test_data_neg_operator(data):
+    neg = -data
+    numpy.testing.assert_allclose(
+        neg.to_array(), -data.to_array(), rtol=1e-15
+    )
+
+
+same_shape = st.shared(qst.qobj_shapes())
+
+@given(qst.qobj_datas(shape=same_shape), qst.qobj_datas(shape=same_shape))
+def test_data_add_operator(a, b):
+    result = a + b
+    numpy.testing.assert_allclose(
+        result.to_array(), a.to_array() + b.to_array(), rtol=1e-15
+    )
```

Comment:
You *could* use a `@pytest.mark.parametrize("op", [operator.add, ...])` here, but it's really a matter of taste.

Reply:
I quite like the separate tests.